sorts/strand_sort.py

```diff
@@ -0,0 +1,60 @@
+from functools import partial
+import operator
+
+
+def strand_sort(arr: list, solution: list, _operator: callable):
```

Comment:
This sort would be easier to use (and test) if it returned a list instead of requiring the caller to allocate and pass in a `solution` list.  It would also be cool if `strand_sort()` was defined to work like the Python builtin [`sorted()`](https://docs.python.org/3/library/functions.html#sorted) which has an optional `reverse` parameter that defaults  to `False`.  We don't need to support the `key` parameter of `sorted()` but the `reverse` would make a lot of sense.

Reply:
I am agreed. One question because this implementation needs recursive passing solution array because algorithm depends on deleting items from input source. Should I  avoid recursion?

Reply:
```python
def strand_sort(arr: list, reverse: bool=False, solution: list=None):
    solution = solution or []
    your code goes here...
```
This will allow the caller to just pass in the `arr` but also allows the function to call itself recursively.  Best of both worlds.